xds/internal/xdsclient/xdsresource/unmarshal_cds_test.go

```diff
@@ -1620,3 +1622,181 @@ func (s) TestUnmarshalCluster(t *testing.T) {
 		})
 	}
 }
+
+func (s) TestValidateClusterWithOutlierDetection(t *testing.T) {
+	oldOutlierDetectionSupportEnv := envconfig.XDSOutlierDetection
+	envconfig.XDSOutlierDetection = true
+	defer func() { envconfig.XDSOutlierDetection = oldOutlierDetectionSupportEnv }()
+	odProto := func(od *v3clusterpb.OutlierDetection) *v3clusterpb.Cluster {
+		// Cluster parsing doesn't fail with respect to fields orthogonal to
+		// outlier detection.
+		return &v3clusterpb.Cluster{
+			Name:                 clusterName,
+			ClusterDiscoveryType: &v3clusterpb.Cluster_Type{Type: v3clusterpb.Cluster_EDS},
+			EdsClusterConfig: &v3clusterpb.Cluster_EdsClusterConfig{
+				EdsConfig: &v3corepb.ConfigSource{
+					ConfigSourceSpecifier: &v3corepb.ConfigSource_Ads{
+						Ads: &v3corepb.AggregatedConfigSource{},
+					},
+				},
+			},
+			LbPolicy:         v3clusterpb.Cluster_ROUND_ROBIN,
+			OutlierDetection: od,
+		}
+	}
+	odUpdate := func(od *OutlierDetection) ClusterUpdate {
```

Comment:
Same here. The func returns a `ClusterUpdate`.

Reply:
Switched to odToClusterUpdate.